presto-spi/src/main/java/com/facebook/presto/spi/security/SslContextProvider.java

```diff
@@ -0,0 +1,284 @@
+        // If a custom truststore is provided but empty, that's an error
+        if (truststore != null) {
+            try {
+                // Check if truststore has any certificates
+                List<String> aliases = Collections.list(truststore.aliases());
+                if (aliases.isEmpty()) {
+                    throw new GeneralSecurityException("Truststore is empty - no trusted certificates found");
+                }
+                log.debug("Truststore contains {} certificate(s): {}", aliases.size(), aliases);
+            }
+            catch (KeyStoreException e) {
+                throw new GeneralSecurityException("Failed to read truststore", e);
+            }
+        }
+
+        trustManagerFactory.init(truststore);
+
+        TrustManager[] trustManagers = trustManagerFactory.getTrustManagers();
+        if (trustManagers.length != 1 || !(trustManagers[0] instanceof X509TrustManager)) {
+            throw new RuntimeException("Unexpected default trust managers: " + Arrays.toString(trustManagers));
+        }
+
+        return (X509TrustManager) trustManagers[0];
+    }
+
+    private static KeyStore loadTrustStore(File trustStorePath, Optional<String> trustStorePassword)
+            throws IOException, GeneralSecurityException
+    {
+        KeyStore trustStore = getInstance(getDefaultType());
+        boolean loaded = false;
+        Exception lastException = null;
+
+        // First try to load as PEM format
+        try {
+            log.debug("Attempting to load truststore as PEM format");
+            List<X509Certificate> certificateChain = PemReader.readCertificateChain(trustStorePath);
+            if (!certificateChain.isEmpty()) {
+                trustStore.load(null, null);
+                for (int i = 0; i < certificateChain.size(); i++) {
+                    X509Certificate certificate = certificateChain.get(i);
+                    X500Principal principal = certificate.getSubjectX500Principal();
+                    String alias = "cert-" + i + "-" + principal.getName();
+                    trustStore.setCertificateEntry(alias, certificate);
+                }
+                loaded = true;
+                log.debug("Successfully loaded {} certificates from PEM truststore", certificateChain.size());
+            }
+        }
+        catch (IOException | GeneralSecurityException e) {
+            log.debug("Failed to load truststore as PEM format: {}", e.getMessage());
+            lastException = e;
+        }
+
+        // If PEM loading failed, try standard KeyStore format
+        if (!loaded) {
+            try {
+                log.debug("Attempting to load truststore as JKS format");
+                try (InputStream inputStream = new FileInputStream(trustStorePath)) {
+                    trustStore.load(inputStream, trustStorePassword.map(String::toCharArray).orElse(null));
+                }
+                loaded = true;
```

Comment:
setting this to true here is redundant, it is not getting used further. Lets remove.